parser: remove experimental multi-region locality syntaxes

otan opened the pull request:
![destroy 75% of the multi-region locality syntaxes](https://user-images.githubusercontent.com/3646147/111238264-dbe74200-864a-11eb-89aa-b1f85a4f84ed.png)

Release note (sql change): Removed the experimental multi-region
locality syntaxes.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62055)
<!-- Reviewable:end -->

ajstorm reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> , but please hold off on merging until we discuss this in the weekly later today.

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62055#-:-MVvC-H0DWvv1WuliPmn:bw3rfdi)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @awoods187)



<!-- Sent from Reviewable.io -->

otan committed:
parser: remove experimental multi-region locality syntaxes

Release note (sql change): Removed the experimental multi-region
locality syntaxes.

otan commented:
modified as discussed in weekly

otan commented:
bors r=ajstorm
